### PR TITLE
Release 0.5.0-beta

### LINKED ISSUE #181: Install OS progress goes to 100% but need more time to complete
**Describe the bug**
Install OS step has a progress indicator going to 100%
You get to a point where the progress bar is at 100% without the user being able to continue and with no progress spinner. The UI seems frozen to the user whereas the user has to wait a little bit longer.

**To Reproduce**
1. Go to install os
2. Wait until 100%
3. See no option yet to continue 
4. See no other indicator of activity 

**Expected behavior**
Either keep progress bar at 99% or something until able to continue, or change to a spinner once bar is at 100%. As to have a visual clue to wait.

**Desktop (please complete the following information):**
 - OS: Debian 10
 - Version 0.4.3-beta

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy S9
 - OS: LineageOS

**Additional context**
-

### PR DESCRIPTION
**Pull request by tsterbak**
This release adds:
- Allow flashing a custom recovery with a toggle switch.
- Display device specific notes at the image selection step (thanks to @anon1892)
- Include the commands `fastboot_flash_recovery` and `fastboot_reboot_recovery` (thanks to @anon1892)
- Add support for Samsung Galaxy S5 LTE (klte).
- Add support for Mi439 (Redmi 7A & co.) (thanks to @anon1892)
- Add support for lavender (Xiaomi Redmi Note 7) (thanks to @anon1892)
- Add support for ginkgo (Xiaomi Redmi Note 8 / 8T) (thanks to @anon1892)
- Add support for garden (Xiaomi Redmi 9A & co.) (thanks to @anon1892)
- Add support for rosemary (Xiaomi Redmi Note 10S & co.) (thanks to @anon1892)
- Add support for miatoll (Xiaomi Redmi Note 9 Pro / 9S ... Poco M2 Pro) (thanks to @anon1892)
- Add support for davinci (Xiaomi Mi 9T / Redmi K20) (thanks to @anon1892)
- Add support for raphael (Xiaomi Mi 9T Pro / Redmi K20 Pro) (thanks to @anon1892)
- Add support for vayu (Poco X3 Pro) (thanks to @anon1892)
- Add support for cupid (Xiaomi 12) (thanks to @anon1892)

Fixes:
- make fastboot_unlock_critical properly accessible.
- Add missing alternative device codes for Samsung S9 and S10 (thanks to @MagicLike)
- Fix the bug with the progress bar percentages (resolves #181)
- Display the scroll bar by default if scrolling is necessary
- An info text box for additional images

Other news:
- Actions for releases are upgraded (by @MagicLike)
- Reworked the device request template as form (by @MagicLike)
- Updated bootloader unlock info (by @freechelmi)

Todo:
- [x] Test flashing with already installed recovery.
- [x] Add new devices to the README
- [x] Update the website with new devices

### CODE REVIEW

**Comment by tsterbak** [comment]
> Why don't you add https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/222 https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/223 https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/224 https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/225 https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/228 https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/234 https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/235 https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/271 and https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/287?
Some are tested, some aren't, but they should be working and don't require more than https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/226 you added here... (just need to remove recovery metadata field, but else it should be work with this version)

I would be happy to add those! Currently, they require orangefox support. If this is dropped for now, I would be happy to add them :)

**Comment by anon1892** [comment]
I don't have the time to do that for now, but just remove these lines :
```
  supported_recovery:
    - orangefox
    - twrp
```
They will work with TWRP too. IIRC, all needed commands were merged for #226 to be working (`fastboot_flash_recovery` and so on)

**Comment by tsterbak** [comment]
This is most likely the final state for the release planed on Friday. Feel free to test it or suggest things that are missing :)
Thanks for everyone involved so far!

**Comment by anon1892** [comment]
Wow, happy to see all these changes being merged!
There is occurrence of `untested` optional metadata in README, but not in device confs neither in code : I checked too fast or it was forgotten in the process? (The code just shows a warning when the device is untested...)

**Comment by tsterbak** [comment]
>  There is occurrence of untested optional metadata in README, but not in device confs neither in code : I checked too fast or it was forgotten in the process? (The code just shows a warning when the device is untested...)

I removed it from the configs, because the config validation failed. But good point, I will better fix the validation and put the metadata field back, where required :)

**Comment by tsterbak** [comment]
> Will you fix it before the release or will it be part of 0.4.6 / a hotfix?

I will fix it for this release.

**Comment by MagicLike** [comment]
> > Will you fix it before the release or will it be part of 0.4.6 / a hotfix?
> 
> I will fix it for this release.

Then I will not approve the configs where a untested tag is needed :+1:

**Comment by MagicLike** [comment]
> > Will you fix it before the release or will it be part of 0.4.6 / a hotfix?
> 
> I will fix it for this release.

Should we wait with the relases? https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/287#issuecomment-1737812931

**Review by MagicLike — COMMENTED** [review]
LGTM - just waiting for the `untested` metadata to get fixed for `cupid`, `davinci`, `ginko`, `miatoll`, `raphael`, rosemary and `vayu`.

**Comment by tsterbak** [comment]
> @tsterbak Should we wait with the release for this? https://github.com/openandroidinstaller-dev/openandroidinstaller/pull/287#issuecomment-1737812931

No, it think we can go out like this. This will also be a test for the "untested" metadata. If we start supporting untested devices, I guess this is the way to go :)

**Review by MagicLike — APPROVED** [review]

**Comment by RaptaG** [comment]
Are the upcoming devices tested?

File: docs/unlocking_the_bootloader.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Devices by *Samsung, Google, OnePlus, Fairphone* and others make it relatively easy to unlock the bootloader and get good support in the installer.
+
+For some devices, it is necessary to unlock the bootloader manually. You usually need to create an account on the manufacturer's website, wait for a certain time and obtain a code from there. OpenAndroidInstaller will try to guide you as far as possible. These vendors include *Sony, Motorola and Xiaomi* amongst others.
+
+Other phone manufacturers no longer allow you to unlock the bootloader. Nothing can be done if you have not unlocked your device in time. These manufacturers include *Huawei, Honor, LG and ASUS*. Support for these manufacturers will always be very limited.
+
+<details>
+    <summary>
+        <b>Here is a brief overview of the bootloader policies of some of the most popular brands (may be incomplete/inaccurate)</b>
+    </summary>
```
Comment (RaptaG) [inline docs/unlocking_the_bootloader.md:14]: The details here could be excluded, it would be more clear to see the instructions that way

**Comment by anon1892** [comment]
> Are the upcoming devices tested?

Some are, some aren't, but behaves the same way others tested device do. See `untested` metadata.

**Comment by RaptaG** [comment]
> Some are, some aren't, but behaves the same way others tested device do. See `untested` metadata.

I see, `miatoll` (my device) is untested sadly and I fear in case of something bad happens, I don't wanna risk my main device.. Does anybody here who has this model (and is willing to) test it?

File: docs/unlocking_the_bootloader.md (hunk 1 above)
Comment (MagicLike) [reply docs/unlocking_the_bootloader.md:14]: Yes, but it would also bloat the page, as the fully detailed list is already linked.

Comment (RaptaG) [reply docs/unlocking_the_bootloader.md:14]: It is not in English, therefore being less accessible, I recommend you reconsider that

Comment (tsterbak) [reply docs/unlocking_the_bootloader.md:14]: Yeah, I also don't like, that its not in English... Maybe just showing the table would be better.

Comment (MagicLike) [reply docs/unlocking_the_bootloader.md:14]: > Yeah, I also don't like, that its not in English... Maybe just showing the table would be better.

hmmm ok - can you please fix that quickly, as I can't commit to dev directly

Comment (tsterbak) [reply docs/unlocking_the_bootloader.md:14]: done

File: docs/unlocking_the_bootloader.md [hunk 2]
```diff
@@ -1,17 +1,14 @@
-#### On unlocking the bootloader
+# On unlocking the bootloader
 
-First of all, this tool will not help you bypass any vender locked bootloaders in non-official ways.
+[!IMPORTANT] First of all, this tool will not help you bypass any vender locked bootloaders in non-official ways.
```
Comment (MagicLike) [inline docs/unlocking_the_bootloader.md:3]: To get the correct formatting, it should be like this:

```
> [!IMPORTANT]
> First of all, this tool will not help you bypass any vender locked bootloaders in non-official ways.
```

Comment (tsterbak) [reply docs/unlocking_the_bootloader.md:3]: I'll fix it

**Review by MagicLike — APPROVED** [review]

**Comment by tsterbak** [comment]
It is released now! :confetti_ball: 
Also, I decided to release it as 0.5.0-beta, since quite a lot of things were added and I thought following semantic versioning a bit better would be good. 

Great work everyone involved! :)

**Pull request merged**